src/DotNetWorker/Converters/ArrayConverter.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.Functions.Worker.Converters
+{
+    // Converting IEnumerable<> to Array
+    internal class ArrayConverter : IConverter
+    {
+        // Convert IEnumerable to array
+        public bool TryConvert(ConverterContext context, out object? target)
+        {
+            // Ensure requested type is an array
+            if (context.Parameter.Type.IsArray)
+            {
+                Type? elementType = context.Parameter.Type.GetElementType();
+                if (elementType is not null)
+                {
+                    // Ensure that we can assign from source to parameter type
+                    if (elementType.IsAssignableFrom(typeof(string))
```

Comment:
All of the comparisons here should be for equality, no need for `IsAssignableFrom`. Types like ReadOnlyMemory<byte>, for example, would only apply in exact matches.